### DISCUSSION TITLE
Request: Add my CLI Tool, Ozed

### DISCUSSION THREAD

**Discussion by dylansantwani** [zed-industries/zed#22405] [Show and tell]
### Check for existing issues

- [X] Completed

### Describe the feature

Hi,

I wrote a cli application called ozed that opens the current directory within zed. It is a compiled bash script that works on linux. Ozed.sh is below.

#!/bin/bash
zed -n $(pwd)

Adding this to path, assuming you have the zed cli installed, allows you to execute ozed anywhere and open the current directory in zed.

### Zed Version and System Specs

Zed: v0.166.1 (Zed)
OS: Linux X11 arch unknown
Memory: 15.3 GiB
Architecture: x86_64
GPU: Intel(R) Arc(tm) Graphics (MTL) || Intel open-source Mesa driver || Mesa 24.3.2-arch1.1

### If applicable, add mockups / screenshots to help present your vision of the feature

![image](https://github.com/user-attachments/assets/1abb8362-4d90-4d28-9616-b0df1db6d019)

**Comment by SomeoneToIgnore** [comment]
Is that all the thing requested to be added?

```
#!/bin/bash
zed -n $(pwd)
```

I see zero value of adding that, as Zed CLI already allows to run `zed -n .` and `#!/bin/bash` shebang odd and not very cross-platform (why would it even need `bash` to start with...).

Also, Zed is written in Rust not in a scripting language, so it's very unclear where would you want to add this.

Also, such requests should better go as Pull Requests https://github.com/zed-industries/zed/pulls instead of all this.

**Reply by dylansantwani** [reply]
Not the CLI. Or they could just make it so that running zed opens the current dir with the script. The script can do the same thing pretty much on win with the replacement of pwd as cd.

**Comment by pressdarling** [comment]
Why not just add a shell alias instead of a script for such a one-liner? You could bind e.g. `zo` to `zed .` and `zn` to `zed -n .` in two lines. I just added this to my `.zshrc` to test:

```zsh
alias zo="zed ."
alias zn="zed -n ."
```

This works with both `/usr/local/bin/zed` and `/opt/homebrew/bin/zed` (macOS) since PATH resolution handles the binary location. The approach works in other shells too - for bash users just add to `.bashrc` (if sourced in `.bash_profile`) and fish users can add to `config.fish`. No need to spawn another shell.

Since you're using OMZ, which doesn't currently have a zed plugin, this could actually be a great opportunity to create one - it could handle cross-platform paths (especially for Windows users where `pwd` vs `%CD%` matters), add command completion, and support more features. Happy to help if you'd like to take that approach instead.
